### PR TITLE
Speed up contribution results by removing join on civicrm_financial_type table when rendering search results.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Speed up contribution results by removing join on civicrm_financial_type table when rendering search results.

Before
----------------------------------------
Join on civicrm_financial_type used to get financial type name = slower

After
----------------------------------------
Pseudoconstant used - faster

Technical Details
----------------------------------------
This can be done much more efficiently through the pseudoconstant mechanism.

Although it is an indexed join it is chosen at the expense of another index - so
your search on payment_instrument_id may not use an index on the filter field
due to this join.

Comments
----------------------------------------

Note that there are multiple unit tests covering filtering by & returning of financial type. Last time I worked in this part of the code I added the function api_v3_ContactTest::testPseudoFields to cover the pseudofields handled at that time. We hit some regressions around order by in search builder so this time I added 

CRM_Contribute_BAO_QueryTest::testSearchPseudoReturnProperties

in preparation for this change

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@monishdeb @jitendrapurohit @seamuslee001 @colemanw really looking for someone who has time to help get this merged - I'd really like it to be in the 5.12 rc

**Comment by jitendrapurohit** [comment]
Tested the search forms with different combination of filters. Seems to work fine on all of them. Query formation before the PR -

    SELECT  ....., civicrm_financial_type.id as financial_type_id, civicrm_financial_type.name as financial_type  

    FROM civicrm_contact contact_a 
        LEFT JOIN civicrm_contribution ON civicrm_contribution.contact_id = contact_a.id  
        INNER JOIN civicrm_financial_type ON civicrm_contribution.financial_type_id = civicrm_financial_type.id  

    WHERE  ...

After -

    SELECT  ..., civicrm_contribution.financial_type_id as `financial_type_id`  
    FROM civicrm_contact contact_a 
        LEFT JOIN civicrm_contribution ON civicrm_contribution.contact_id = contact_a.id  
    WHERE ...

The output rows are same and unaffected. No errors noticed on any forms after applying the PR. Looks good to be merged 👍

**Review by jitendrapurohit — APPROVED** [review]

File: xml/schema/Contribute/Contribution.xml [hunk 1]
```diff
@@ -77,7 +77,7 @@
   </foreignKey>
   <field>
     <name>financial_type_id</name>
-    <title>Financial Type</title>
+    <title>Financial Type ID</title>
```
Comment (colemanw) [inline xml/schema/Contribute/Contribution.xml]: Why this change?

Comment (eileenmcnaughton) [reply xml/schema/Contribute/Contribution.xml]: We have done this with a few fields - it differentiates them in search builder. I'm OK with leaving out if you want
